addons/iot_drivers/server_logger.py

```diff
@@ -1,171 +1,129 @@
+from odoo.addons.iot_drivers.tools.system import IOT_IDENTIFIER
 from odoo.netsvc import ColoredFormatter
 
 _logger = logging.getLogger(__name__)
 
-IOT_LOG_TO_SERVER_CONFIG_NAME = 'iot_log_to_server'  # config name in odoo.conf
-
-
-class AsyncHTTPHandler(logging.Handler):
-    """
-    Custom logging handler which send IoT logs using asynchronous requests.
-    To avoid spamming the server, we send logs by batch each X seconds
-    """
-    _MAX_QUEUE_SIZE = 1000
-    """Maximum queue size. If a log record is received but the queue if full it will be discarded"""
-    _MAX_BATCH_SIZE = 50
-    """Maximum number of sent logs batched at once. Used to avoid too heavy request. Log records still in the queue will
-    be handle in future flushes"""
-    _FLUSH_INTERVAL = 0.5
-    """How much seconds it will sleep before checking for new logs to send"""
-    _REQUEST_TIMEOUT = 0.5
-    """Amount of seconds to wait per log to send before timeout"""
-    _DELAY_BEFORE_NO_SERVER_LOG = 5 * 60  # 5 minutes
-    """Minimum delay in seconds before we log a server disconnection.
-    Used in order to avoid the IoT log file to have a log recorded each _FLUSH_INTERVAL (as this value is very small)"""
-
-    def __init__(self, odoo_server_url, active):
-        """
-        :param odoo_server_url: Odoo Server URL
+
+@helpers.require_db
+class ServerLogger(logging.Handler):
+    _flush_thread: threading.Thread = None
+    _previous_disconnection: float = 0.0
+    _MAX_QUEUE_SIZE: int = 1000
+
+    def __init__(self, server_url=None):
+        """Custom logging handler which send IoT logs using asynchronous requests.
+        To avoid spamming the server, we send logs by batch each X seconds
+
+        :param str server_url: URL of the Odoo server (provided by decorator).
         """
         super().__init__()
-        self._odoo_server_url = odoo_server_url
+        self.setFormatter(
+            ColoredFormatter('%(asctime)s %(pid)s %(levelname)s %(dbname)s %(name)s: %(message)s %(perf_info)s')
+        )
+        self._server_iot_log_url = server_url + '/iot/log'
+        self.odoo_session = requests.Session()
         self._db_name = system.get_conf('db_name') or ''
-        self._log_queue = queue.Queue(self._MAX_QUEUE_SIZE)
-        self._flush_thread = None
-        self._active = None
-        self._next_disconnection_time = None
-        self.toggle_active(active)
-
-    def toggle_active(self, is_active):
+        self.odoo_session.headers = {
+            'X-Odoo-Database': self._db_name,
+            'User-Agent': 'OdooIoTBox/1.0',
+        }
+        self._queue = queue.Queue(self._MAX_QUEUE_SIZE)
+        self._active = True
+        self._flush_thread = threading.Thread(target=self.share_logs_loop, name="ThreadServerLogger", daemon=True)
+        self._flush_thread.start()
+
+    def share_logs_loop(self):
+        while self._active:  # allow to exit the loop on thread.join
+            time.sleep(12)  # sleep before sending logs
```

Comment:
So you have increased the delay between batches from 0.5s to 12s, and not changed the batch size?